src/index.ts

```diff
@@ -46,51 +47,122 @@ declare global {
 }
 
 /**
- * A function to begin a mode to draw cuts.
- * If atomMode was previously active, remove the listener.
+ * If there is no current mode creates the listeners.
+ * Sets the current mode to cut mode.
  */
 function ellipseMode() {
-    if (modeState !== "ellipseMode") {
-        removeListeners();
-        modeState = "ellipseMode";
+    if (modeState === "") {
+        window.addEventListener("keypress", keyPressHandler);
+        canvas.addEventListener("mousedown", mouseDownHandler);
+        canvas.addEventListener("mousemove", mouseMoveHandler);
+        canvas.addEventListener("mouseup", mouseUpHandler);
+        canvas.addEventListener("mouseout", mouseOutHandler);
     }
-    canvas.addEventListener("mousedown", cutHandler);
-    canvas.addEventListener("mousemove", cutHandler);
-    canvas.addEventListener("mouseup", cutHandler);
-    canvas.addEventListener("mouseout", cutHandler);
+    modeState = "cutMode";
 }
 
 /**
- * A function to begin atom creation.
- * If ellipseMode was previously active, remove the listener.
+ * If there is no current mode creates the listeners.
+ * Sets the current mode to atom mode.
  */
 function atomMode() {
-    if (modeState !== "atomMode") {
-        removeListeners();
-        modeState = "atomMode";
+    if (modeState === "") {
+        window.addEventListener("keypress", keyPressHandler);
+        canvas.addEventListener("mousedown", mouseDownHandler);
+        canvas.addEventListener("mousemove", mouseMoveHandler);
+        canvas.addEventListener("mouseup", mouseUpHandler);
+        canvas.addEventListener("mouseout", mouseOutHandler);
     }
-    window.addEventListener("keypress", atomHandler);
-    canvas.addEventListener("mousedown", atomHandler);
-    canvas.addEventListener("mousemove", atomHandler);
-    canvas.addEventListener("mouseup", atomHandler);
-    canvas.addEventListener("mouseout", atomHandler);
+    modeState = "atomMode";
 }
```

Comment:
The assignment of the UEHs has nothing to do with the mode anymore, its completely decoupled, just move their assignment to the global scope right after you assign all the global vars and remove them from the mode button press.
```ts
window.addEventListener("keypress", keyPressHandler);
canvas.addEventListener("mousedown", mouseDownHandler);
canvas.addEventListener("mousemove", mouseMoveHandler);
canvas.addEventListener("mouseup", mouseUpHandler);
canvas.addEventListener("mouseout", mouseOutHandler);
```

Reply:
Sorry about that, was trying to avoid globals without really thinking about it.

Reply:
its good to avoid globals, we're not using best practices here, all of this logic should eventually be moved to a static context object, but for now its fine to introduce globals sparingly if they truly represent a part of global application state such as the mouse being down etc.